### DISCUSSION TITLE
FPGA ID code 0xffffffff

### DISCUSSION THREAD

**Discussion by simntonic** [ataradov/usb-sniffer#42] [Q&A] [answered]
Hi all :)

I'm trying to set this up - hand-soldering was difficult but looking at it with a magnifying glass and testing (most) pins seems ok.
The first command ( `--mcu-sram` ) works fine, and it re-enumerates correctly.
Then the second command (`--mcu-eeprom`) says that the FPGA returned a wrong ID code: 0xffffffff.

I guess some pins are misoldered, any suggestions on which it could be?

Thanks!

**Comment by ataradov** [comment]
Check FPGA orientation (pin 1) first. Otherwise, there are a lot of pins that may be wrong. You will get this result even if FPGA is not soldered at all, so it is hard to tell.

**Comment by simntonic** [comment]
Wow thanks so much for the quick reply! :) Well I did fight a bit with the orientation since there are 2 dots ( >_< ) on the FPGA and nowhere in the doc I could find which one was pin 1. But I saw in the discussions here where you said the small dot is the 1.

Here is a picture
<img width="866" alt="image" src="https://github.com/user-attachments/assets/332c832d-1ef0-48d9-a93a-bd936e1744af">

When I power the board one of the 4 "DGB" LEDs attached to the FPGA turns on, the third one from the left (looking at it on the picture above)

**Comment by ataradov** [comment]
Orientation is correct. All JTAG pins are on the top side (where IC1 is marked). They are the top 4 traces going from IC9 (pins 90, 91, 94, 95 of the FPGA). Pay close attention to those. But also double check soldering on all other pins.

LEDs are fine, they don't mean anything on the unprogrammed FPGA. This is just the default state of the pins.

**Comment by simntonic** [comment]
Ok thank you that helps a lot - one of the 4 pins was not soldered properly on the MCU.
One thing to note: the pads on the MCU footprint are very short, especially for something that needs to be hand soldered. I'd highly recommend anyone making this board to expand those pads for less soldering errors :) 

Now I could program the FPGA and everything seemed ok but when I run the `./usb_sniffer --test` command I get transfer rate speed of 0, just like #33 - I will check the pins of the USB PHY...
(I already checked the 60MHz clock which is ok)

**Comment by ataradov** [comment]
If the 60 MHz is there, then PHY is fine at this stage. Fro the test 60 MHz is all you need from the PHY.

Check other MCU pins. There are a lot of them that would lead to 0 on a speed test.

**Comment by simntonic** [comment]
Ahhh almost there - now finally the test is working, and wireshark is up, but in the capture I get either USB PHY Error or WTAP_ENCAP?

Thank you :)

**Comment by ataradov** [comment]
Did you configure the capture speed correctly for your device?

**Reply by simntonic** [reply]
Thanks for the reply :) 

It works now, the problem was that I was using wireshark 3.6 and not 4.x.

But only full-speed works, high-speed shows "USB PHY Error" -> how do you configure the capture speed?

**Comment by ataradov** [comment]
See Usage section of the min README file.

**Comment by simntonic** [comment, marked as answer]
Awesome - even easier than I thought - all working now thank you!!
Much appreciated for you to give all of this for free :))

If anyone else is reading this: literally ALL the issues I had were related to the hand-soldering, I would heavily recommend to solder as few parts as possible, ideally get a PCBA house (for example JLC) to assemble most of it, 100% worth it!
And if you want to hand-solder, change the pad of the MCU to make them longer ;)
